### PR TITLE
feat: enable compression of docs stored on S3

### PR DESCRIPTION
**Pull request by jcrossley3**
Fixes: #925

### CODE REVIEW

File: modules/storage/src/service/compression/mod.rs [hunk 1]
```diff
@@ -22,6 +23,17 @@ impl Display for Compression {
     }
 }
 
+impl FromStr for Compression {
```
Comment (ctron) [inline modules/storage/src/service/compression/mod.rs:26]: I think we already use `strum`, which should be able to do the same.

File: modules/storage/src/service/s3.rs [hunk 2]
```diff
@@ -47,11 +60,18 @@ impl StorageBackend for S3Backend {
...
+            .compress(file.reader().await.map_err(errhdlr)?)
+            .await;
         let result = file.result();
 
         self.bucket
-            .put_object_stream(&mut source, result.key().to_string())
+            .put_object_stream_with_content_type(
+                &mut source,
+                result.key().to_string(),
+                "application/json",
```
Comment (ctron) [inline modules/storage/src/service/s3.rs:73]: Does that require all documents to be JSON? IIRC, that's not the case for some OSV documents, which are YAML instead.

**Review by ctron — COMMENTED** [review]
Looks good. Two minor things to sort out.

File: modules/storage/src/service/compression/mod.rs (hunk 1 above)
Comment (jcrossley3) [reply modules/storage/src/service/compression/mod.rs]: Updated. Got a little carried away handling the potential `ParseError` though.

File: modules/storage/src/service/s3.rs (hunk 2 above)
Comment (jcrossley3) [reply modules/storage/src/service/s3.rs]: Good call. I realized we don't even use the content-type, so the default will do. Besides, I don't think we want to ripple a content-type argument through our existing storage api.

**Review by ctron — APPROVED** [review]

**Pull request merged**